File: JianZhi_offer_67/code_cpp/7.cpp


class Solution {
public:
    int Fibonacci(int n) {
        int *F = new int[40];
        F[0] = 0;
        F[1] = 1;
        for(int i = 2;i<=n;i++)
        {
            F[i] = F[i-1]+F[i-2];
        }
        return F[n];
    }
};

File: leetcode_learning/code_py/7.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        sum = 0
        temp = 0
        if x>0:
            while(x!=0):
                temp = x % 10
                sum = sum*10 + temp
                x = x // 10
            if (sum > 2**31-1) or (sum < -2**31):
                return 0
            return sum
        elif x<0:
            y = x*(-1)
            while(y!=0):
                temp = y % 10
                sum = sum*10 + temp
                y = y // 10
            sum_minus = sum*(-1)
            if (sum_minus > 2**31-1) or (sum_minus < -2**31):
                return 0
            return sum_minus
        else:
            return 0


x = 1001
s = Solution()
print(s.reverse(x))
